rpcs3/rpcs3qt/emu_settings.cpp

```diff
@@ -146,25 +152,71 @@ emu_settings::Render_Creator::Render_Creator()
 #endif
 
 #if defined(WIN32) || defined(HAVE_VULKAN)
-	// check for vulkan adapters
-	vk::context device_enum_context;
-	u32 instance_handle = device_enum_context.createInstance("RPCS3", true);
+	// Some drivers can get stuck when checking for vulkan-compatible gpus, f.ex. if they're waiting for one to get
+	// plugged in. This whole contraption is for showing an error message in case that happens, so that user has
+	// some idea about why the emulator window isn't showing up.
+
+	static std::atomic<bool> was_called = false;
+	if (was_called.exchange(true))
+		fmt::throw_exception("Render_Creator cannot be called more than once" HERE);
+
+	static std::mutex mtx;
+	static std::condition_variable cond;
+	static bool thread_running = true;
+	static volatile bool device_found = false;
 
-	if (instance_handle > 0)
+	static QStringList compatible_gpus;
+
+	named_thread enumerator_thread("Vulkan device enumeration", [&]
 	{
-		device_enum_context.makeCurrentInstance(instance_handle);
-		std::vector<vk::physical_device>& gpus = device_enum_context.enumerateDevices();
+		thread_ctrl::set_native_priority(-1);
+
+		vk::context device_enum_context;
+		u32 instance_handle = device_enum_context.createInstance("RPCS3", true);
 
-		if (!gpus.empty())
+		if (instance_handle > 0)
 		{
-			//A device with vulkan support found. Init data
-			supportsVulkan = true;
+			device_enum_context.makeCurrentInstance(instance_handle);
+			std::vector<vk::physical_device> &gpus = device_enum_context.enumerateDevices();
 
-			for (auto& gpu : gpus)
+			if (!gpus.empty())
 			{
-				vulkanAdapters.append(qstr(gpu.get_name()));
+				device_found = true;
+
+				for (auto &gpu : gpus)
+				{
+					compatible_gpus.append(qstr(gpu.get_name()));
+				}
 			}
 		}
+		std::scoped_lock{mtx}, thread_running = false;
+		cond.notify_all();
+	});
+
+	{
+		std::unique_lock lck(mtx);
+		// Loop is used to prevent sprudious wakeups from triggering false positives
+		while (thread_running && cond.wait_for(lck, std::chrono::seconds(10)) != std::cv_status::timeout);
```

Comment:
Honestly, for 10s timeout it shall use second overload of wait_for function with predicate.